### PR TITLE
fix bip109 minimum version criteria

### PR DESCRIPTION
**Pull request by ftrader**
Proposed fix for incorrect bip109 tallying in getblockchaininfo RPC.

### CODE REVIEW

File: src/test/rpc_tests.cpp [hunk 1]
```diff
@@ -308,4 +343,205 @@ BOOST_AUTO_TEST_CASE(rpc_ban)
     BOOST_CHECK_EQUAL(adr.get_str(), "2001:4d48:ac57:400:cacf:e9ff:fe1d:9c63/128");
 }
 
+
+BOOST_AUTO_TEST_CASE(rpc_getblockchaininfo)
+{
+    // test bip109 counting
+
+    // need the chain params for setting up test chains and cases
+    SelectParams(CBaseChainParams::MAIN);
```
Comment (zander) [inline src/test/rpc_tests.cpp]: Maybe you want to use testnet.  It would make the test run much faster.

File: src/test/rpc_tests.cpp [hunk 2]
```diff
@@ -308,4 +343,205 @@ BOOST_AUTO_TEST_CASE(rpc_ban)
...
+    
+    // ensure that the hard fork is not yet seen as active
+    // otherwise the global pblocktree must be set up right,
+    // or else an assert will trip in HardForkMajorityDesc
+    sizeForkTime.store(std::numeric_limits<uint32_t>::max());
+    
+    ///////////////////////////////////////////////////////////////////////////
+    // case 1: a block history with only version 4 blocks (zero bip109 found)
+    std::vector<CBlockIndex> *testblocks;
+    testblocks = freshchain(chain_len);
```
Comment (zander) [inline src/test/rpc_tests.cpp]: I guess we don't need chain_len everywhere, just 3 or so would hit all the code and actually test what we want to test, no?

File: src/test/rpc_tests.cpp [hunk 3]
```diff
@@ -308,4 +343,205 @@ BOOST_AUTO_TEST_CASE(rpc_ban)
...
+{
+    // test bip109 counting
+
+    // need the chain params for setting up test chains and cases
+    SelectParams(CBaseChainParams::MAIN);
+    const Consensus::Params& params = Params().GetConsensus();
+    // we need 2 more than the size of the majority window because
+    // the last block (tip) does not count (ref. "blocks" entry
+    // listing height as 1 less than the size of the chain)
+    //const uint32_t chain_len = params.nMajorityWindow + 1;
```
Comment (pdpi) [inline src/test/rpc_tests.cpp]: dead code in comment?

File: src/test/rpc_tests.cpp [hunk 4]
```diff
@@ -308,4 +343,205 @@ BOOST_AUTO_TEST_CASE(rpc_ban)
...
 
+
+BOOST_AUTO_TEST_CASE(rpc_getblockchaininfo)
+{
+    // test bip109 counting
+
+    // need the chain params for setting up test chains and cases
+    SelectParams(CBaseChainParams::MAIN);
+    const Consensus::Params& params = Params().GetConsensus();
+    // we need 2 more than the size of the majority window because
```
Comment (pdpi) [inline src/test/rpc_tests.cpp]: we need 1 more, no?

File: src/test/rpc_tests.cpp (hunk 3 above)
Comment (ftrader) [inline src/test/rpc_tests.cpp]: oops - thanks for spotting!

File: src/test/rpc_tests.cpp (hunk 4 above)
Comment (ftrader) [inline src/test/rpc_tests.cpp]: you are correct, that comment is obsolete and based on earlier misunderstanding of mine

**Comment by ftrader** [comment]
added fixes from previous review comments:
- use testnet
- introduce a min_chain_len for test which does not need full window
- fix obsolete comments / dead code
- remove obsolete case #5a which was a dup of #2, rename #5b to #5

**Comment by zander** [comment]
Great fix, thanks!

**Pull request merged**